CenterForOpenScience/osf.io
# [PLAT-486] Allow drag-and-drop local image upload to wiki

## Purpose

Return the ability to drop images into the Wiki with added information about user permissions and error handling.

## Changes

- Reverts ability to drag things into the wiki
- Adds new indicator if user has ability to upload files through dragging.
- Adds new error messages alerting the user if they try to upload without permission.
- Adds new error messages alerting the user if they try overwrite an existing file.

Can't rename Wiki images messages
<img width="1439" alt="screen shot 2018-02-16 at 11 33 16 am" src="https://user-images.githubusercontent.com/9688518/36318829-05d118d2-130f-11e8-82f2-5cba7efea974.png">

Can't overwrite Wiki images messages
<img width="1440" alt="screen shot 2018-02-16 at 11 33 38 am" src="https://user-images.githubusercontent.com/9688518/36318830-05defdee-130f-11e8-87a0-6bddbd029e67.png">

Delete Wiki images folder 
<img width="1440" alt="screen shot 2018-02-16 at 11 34 05 am" src="https://user-images.githubusercontent.com/9688518/36318831-061c4104-130f-11e8-925f-8c31c036c883.png">

Delete file in Wiki images folder
<img width="1437" alt="screen shot 2018-02-16 at 11 34 18 am" src="https://user-images.githubusercontent.com/9688518/36318832-062c4d38-130f-11e8-8a5a-b93917674e08.png">

Delete multiple files in Wiki images folder
<img width="1440" alt="screen shot 2018-02-16 at 11 43 14 am" src="https://user-images.githubusercontent.com/9688518/36318833-063d03da-130f-11e8-805b-fd4adbfeb21a.png">

Upload wrong file type
<img width="1437" alt="screen shot 2018-02-16 at 11 44 41 am" src="https://user-images.githubusercontent.com/9688518/36318836-06702c74-130f-11e8-87da-7171c377d364.png">

Permission error
<img width="1439" alt="screen shot 2018-02-16 at 11 57 10 am" src="https://user-images.githubusercontent.com/9688518/36319334-99bcf4d4-1310-11e8-86bb-a9582efcc46e.png">

## QA Notes

Drag and drop various images with various permission levels and make sure the images are formatted properly. Try dropping many images at once large images etc.

Also includes analytics here's the suggested custom report.

<img width="1439" alt="screen shot 2018-03-12 at 6 00 22 pm" src="https://user-images.githubusercontent.com/9688518/37311894-737266ac-261f-11e8-831b-ac5e2bf1b2b0.png">
 

## Side Effects

Registering a private project will still cause the resulting registrations wiki links to fail, because they will still be pointing at the old private projects files, a fix for this will come later after the wiki remodeling is complete.

## Ticket

https://openscience.atlassian.net/browse/PLAT-486

## Notes 
Revival of #4569
Credit goes to @ticklemepierce
